spec/controllers/sessions_controller_spec.rb

```diff
@@ -3,37 +3,78 @@
 require 'rails_helper'
 describe SessionsController do
   describe '#create' do
+    before do
+      request.env['omniauth.auth'] = OmniAuth::AuthHash.new(
+        {
+            provider: 'google_oauth2',
+            info: { email: 'jake@magmalabs.io' }
+        }
+      )
+    end
+
     context 'with oauth' do
       it 'signs in existing developer by email' do
-        developer = FactoryBot.create :developer, email: 'jake@example.com'
-        request.env['omniauth.auth'] = {
-            'info' => {
-                'email' => 'jake@example.com'
-            }
-        }
+        developer = FactoryBot.create :developer, email: 'jake@magmalabs.io'
 
         expect do
           post :create
           expect(response).to redirect_to root_path
-        end.to change { Authem::Session.count }.by(1)
-        expect(Authem::Session.last.subject_id).to eq developer.id
+        end.to change(session, :count).by(2)
+
+        expect(session[:developer_email]).to eq developer.email
       end
 
-      it 'signs up a new developer by email' do
-        request.env['omniauth.auth'] = {
-            'info' => {
-                'email' => 'newdev@example.com',
-                'name' => 'John Smith'
-            }
-        }
+      it 'does successfully create a session' do
+        expect(session[:developer_id]).to be_nil
+        post :create, params: { provider: :google_oauth2 }
+        expect(session[:developer_email]).not_to be_nil
+      end
+    end
+
+    context 'when signs up' do
+      before do
+        request.env['omniauth.auth'] = OmniAuth::AuthHash.new(
+          {
+              provider: 'google_oauth2',
+              info: {
+                  email: 'newdev@magmalabs.io',
+                  name: 'John Smith'
+              }
+          }
+        )
+      end
 
+      it 'a new developer by email' do
         expect do
           post :create
           expect(response).to redirect_to root_path
         end.to change(Developer, :count).by(1)
         developer = Developer.last
```

Comment:
I tend not to trust in first/last/etc speaking about Postgresql and sql server, they return the rows as soon as they read them from disk, but that does not ensure they are sorted, it depends on internal disk availability